pkgs/by-name/rw/rwpspread/package.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, libxkbcommon
+, stdenv
+, wayland
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rwpspread";
+  version = "0.1.8";
+
+  src = fetchFromGitHub {
+    owner = "0xk1f0";
+    repo = "rwpspread";
+    rev = "v${version}";
+    hash = "sha256-slxsicASZ7JoUnnQf4R3xFB4zgtt4ZOZCU0NcbgBneM=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "smithay-client-toolkit-0.18.0" = "sha256-6y5abqVHPJmh8p8yeNgfTRox1u/2XHwRo3+T19I1Ksk=";
+    };
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    libxkbcommon
+  ] ++ lib.optionals stdenv.isLinux [
+    wayland
+  ];
+
+  meta = with lib; {
```

Comment:
You may want to remove this `with lib` statement and use `lib.` directly — that's [feedback I received in the past](https://github.com/NixOS/nixpkgs/pull/271173#discussion_r1411200326), anyway.

Reply:
`with` is generally ugly imo anyway, may as well.
If only removing `rec` was as clean here :(

Reply:
I think you can use `finalAttrs` for this — I used it in my PR to avoid a `rec`.

Reply:
I would leave the `rec` as is, and remove the `with lib;`

Reply:
unfortunately not here, thank you!